include/cpr/auth.h

```diff
@@ -21,9 +21,11 @@ class Authentication {
     Authentication& operator=(const Authentication& other) = default;
 
     virtual const char* GetAuthString() const noexcept;
+    virtual AuthMode GetAuthMode() const noexcept;
```

Comment:
Since we do not derive from this class any more, we don't need any `virtual` methods here. Also library consumers are not supposed to derive from this class either.

```suggestion
    AuthMode GetAuthMode() const noexcept;
```